Integrate Frgaal in Java libs

hubertp opened the pull request:
[ci no changelog needed]

### Pull Request Description

This PR allows for easier integration of Frgaal in different subprojects.
Related to https://www.pivotaltracker.com/story/show/182093808

### Checklist

Please include the following checklist in your PR:

- [ ] The documentation has been updated if necessary.
- [x] All code conforms to the [Scala](https://github.com/enso-org/enso/blob/develop/docs/style-guide/scala.md), [Java](https://github.com/enso-org/enso/blob/develop/docs/style-guide/java.md), and [Rust](https://github.com/enso-org/enso/blob/develop/docs/style-guide/rust.md) style guides.
- All code has been tested:
  - [ ] Unit tests have been written where possible.
  - [ ] If GUI codebase was changed: Enso GUI was tested when built using BOTH `./run dist` and `./run watch`.

hubertp committed:
Integrate Frgaal in Java libs

Refactored the Frgaal setting to allow for easier integration.

4e6 commented:
@hubertp there are some issues with compilation https://github.com/enso-org/enso/runs/6320685106?check_suite_focus=true#step:4:298

<blockquote><img src="https://opengraph.githubassets.com/79a8c918e7e1d70cf81482237dd8d8e806ed2bd90e54d0c866826c1afe6f47e7/enso-org/enso" width="48" align="right"><div><img src="https://github.githubassets.com/favicons/favicon.svg" height="14"> GitHub</div><div><strong><a href="https://github.com/enso-org/enso">Integrate Frgaal in Java libs · enso-org/enso@b70ec8e</a></strong></div><div>Hybrid visual and textual functional programming. Contribute to enso-org/enso development by creating an account on GitHub.</div></blockquote>

hubertp commented:
@4e6 thanks, I somehow missed it. Sorry

hubertp committed:
Merge branch 'develop' into hubert/wip/more-frgaal-integration-182093808

hubertp committed:
Make sure truffle API is on the classpath

When not using javac, truffle has to be added as an explicit dependency
(still classified as `provided`).

hubertp commented:
@4e6 done

4e6 commented on build.sbt:
```diff
@@ -1589,14 +1597,16 @@ lazy val `std-base` = project
       "org.apache.poi"      % "poi-ooxml"         % "5.0.0",
-      "org.apache.xmlbeans" % "xmlbeans"          % "5.0.1"
+      "org.apache.xmlbeans" % "xmlbeans"          % "5.0.1",
+      "org.graalvm.truffle" % "truffle-api"       % graalVersion % "provided"
```
I wonder how it worked before

4e6 reviewed (approved):
Looks good!

mergify[bot] committed:
Merge branch 'develop' into hubert/wip/more-frgaal-integration-182093808

mergify[bot] committed:
Merge branch 'develop' into hubert/wip/more-frgaal-integration-182093808
